### PR TITLE
[circle-mpqsolver] Revisit  BisectionSolver

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit excludes unnecessary search process when target error is out of reasonable bounds.

Its output for `qerror_ratio = 0.0` (q16) is:
```
...
>> Computing baseline qerrors
Full int16 model qerror: 9.25156e-05
Full uint8 model qerror: 0.000252629
Target qerror: 9.25156e-05
The best configuration is int16 configuration 
```
Its output for `qerror_ratio = 1.0` (q8) is:
```
...
>> Computing baseline qerrors
Full int16 model qerror: 9.25156e-05
Full uint8 model qerror: 0.000252629
Target qerror: 0.000252629
The best configuration is uint8 configuration 
```
ONE-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

**Comment by stamalakhov** [comment]
@jinevening 
Please take another look

**Review by jinevening — DISMISSED** [review]
LGTM

File: compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp [hunk 1]
```diff
@@ -140,16 +140,45 @@ std::unique_ptr<luci::Module> BisectionSolver::run(const std::string &module_pat
...
+      return nullptr;
+    }
+
+    if (_hooks)
+    {
+      _hooks->on_end_solver(layer_params, "int16", int16_qerror);
+    }
+
+    SolverOutput::get() << "The best configuration is int16 configuration "
+                        << "\n";
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp:159]:
```suggestion
    SolverOutput::get() << "The best configuration is int16 configuration\n";
```
?

File: compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp [hunk 2]
```diff
@@ -140,16 +140,45 @@ std::unique_ptr<luci::Module> BisectionSolver::run(const std::string &module_pat
...
       return nullptr;
     }
+
+    if (_hooks)
+    {
+      _hooks->on_end_solver(layer_params, "uint8", uint8_qerror);
+    }
+
+    SolverOutput::get() << "The best configuration is uint8 configuration "
+                        << "\n";
```
Comment (seanshpark) [inline compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp:177]:
```suggestion
    SolverOutput::get() << "The best configuration is uint8 configuration\n";
```
?

Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp:177]: Ahh. Sorry. Thank you.

File: compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp (hunk 1 above)
Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/bisection/BisectionSolver.cpp:159]: Ahh. Sorry. Thank you.

**Review by jinevening — APPROVED** [review]

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**